tests/e2e/Services/Sites/SitesCustomServerTest.php

```diff
@@ -65,6 +65,202 @@ public function testCreateSite(): void
+
+        // get secret variable
+        $secretVariable = $this->getVariable($siteId, $secretVariable['body']['$id']);
+
+        $this->assertEquals(200, $secretVariable['headers']['status-code']);
+        $this->assertNotEmpty($secretVariable['body']['$id']);
+        $this->assertEquals('siteKey3', $secretVariable['body']['key']);
+        $this->assertEquals('', $secretVariable['body']['value']);
+        $this->assertEquals(true, $secretVariable['body']['secret']);
+
+        // list variables
+        $variables = $this->listVariables($siteId);
+
+        $this->assertEquals(200, $variables['headers']['status-code']);
+        $this->assertCount(3, $variables['body']['variables']);
+
+        // delete variable
+        $this->deleteVariable($siteId, $variable['body']['$id']);
+        $this->deleteVariable($siteId, $variable2['body']['$id']);
+        $this->deleteVariable($siteId, $secretVariable['body']['$id']);
+
+        // list variables
+        $variables = $this->listVariables($siteId);
+
+        $this->assertEquals(200, $variables['headers']['status-code']);
+        $this->assertCount(0, $variables['body']['variables']);
+
+        $this->cleanupSite($siteId);
+    }
+
+    public function testVariablesE2E(): void
+    {
+        $siteId = $this->setupSite([
+            'siteId' => ID::unique(),
+            'name' => 'Astro site',
+            'framework' => 'astro',
+            'adapter' => 'ssr',
+            'buildRuntime' => 'ssr-22',
+            'outputDirectory' => './dist',
+            'buildCommand' => 'npm run build',
+            'installCommand' => 'npm install',
+            'fallbackFile' => '',
+        ]);
+
+        $this->assertNotEmpty($siteId);
+
+        $secretVariable = $this->createVariable($siteId, [
+            'key' => 'name',
+            'value' => 'Appwrite',
+        ]);
+
+        $this->assertEquals(201, $secretVariable['headers']['status-code']);
+        $this->assertNotEmpty($secretVariable['body']['$id']);
+        $this->assertEquals('name', $secretVariable['body']['key']);
+        $this->assertEquals('', $secretVariable['body']['value']);
+        $this->assertEquals(true, $secretVariable['body']['secret']);
+
+        $deploymentId = $this->setupDeployment($siteId, [
+            'code' => $this->packageSite('astro'),
+            'activate' => 'true'
+        ]);
+
+        $this->assertNotEmpty($deploymentId);
+
+        $domain = $this->getSiteDomain($siteId);
+        $proxyClient = new Client();
+        $proxyClient->setEndpoint('http://' . $domain);
+
+        $response = $proxyClient->call(Client::METHOD_GET, '/', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ]));
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertStringContainsString("Message from ENV: Appwrite", $response['body']);
```

Comment:
Let's also add assertion that response body does NOT contain `Variable not found`